ecommerce/extensions/catalogue/management/commands/update_course_seat_expire.py

```
@@ -0,0 +1,121 @@
+                    seats_updated += 1
+
+        logger.info("[%d] seats updated", seats_updated)
+        self.log_enrollment_missing()
+
+    def log_enrollment_missing(self):
+        """ log courses which are missing enrollment end date """
+        logger.info("Enrollment missing for: [%d] courses", len(self.enrollment_date_not_found))
+
+        for course_key in self.enrollment_date_not_found:
+            logger.info("%s", course_key)
+
+    def _get_courses_information(self, **querystring):
+        """
+        Retrieve the enrollment information for all the courses.
+
+        Arguments:
+            querystring: keyword args which are passed as querystring
+
+        Returns a key, value pair of (course_key, enrollment_end) course and enrollment info.
+        """
+        def _parse_response(api_response):
+            response_data = api_response.get('results', self.no_data)
+
+            # Map course_id with enrollment end date.
+            courses_enrollment = dict(
+                (course_info['course_id'], course_info['enrollment_end'])
+                for course_info in response_data
+            )
+            return courses_enrollment, api_response['pagination'].get('next', None)
+
+        querystring = {} if not querystring else querystring
+        api = EdxRestApiClient(get_lms_url('api/courses/v1/'))
+        course_enrollments = {}
+        try:
+            page = 1
+            response = api.courses().get(**querystring)
+            enrollment_info, next_page = _parse_response(response)
+            course_enrollments.update(enrollment_info)
+            while next_page:
+                page += 1
+                querystring['page'] = page
+                response = api.courses().get(**querystring)
+                enrollment_info, next_page = _parse_response(response)
+                course_enrollments.update(enrollment_info)
+            return course_enrollments
+
+        except Exception:  # pylint: disable=broad-except
```

Let the exception bubble up. We cannot recover from it.